src/Microsoft.DotNet.Tools.Publish/PublishCommand.cs

```diff
@@ -250,5 +252,71 @@ private static IEnumerable<ProjectContext> GetMatchingProjectContexts(IEnumerabl
                 }
             }
         }
+
+        private static void CopyContents(ProjectContext context, string outputPath)
+        {
+            var contentFiles = context.ProjectFile.Files.GetContentFiles();
+            Copy(contentFiles, context.ProjectDirectory, outputPath);
+        }
+
+        private static void Copy(IEnumerable<string> contentFiles, string sourceDirectory, string targetDirectory)
+        {
+            if (contentFiles == null)
+            {
+                throw new ArgumentNullException(nameof(contentFiles));
+            }
+
+            sourceDirectory = EnsureTrailingSlash(sourceDirectory);
+            targetDirectory = EnsureTrailingSlash(targetDirectory);
+
+            foreach (var sourceFilePath in contentFiles)
+            {
+                Reporter.Verbose.WriteLine($"Publishing {sourceFilePath.Green().Bold()} ...");
+
+                var fileName = Path.GetFileName(sourceFilePath);
+
+                var targetFilePath = sourceFilePath.Replace(sourceDirectory, targetDirectory);
+                var targetFileParentFolder = Path.GetDirectoryName(targetFilePath);
+
+                // Create directory before copying a file
+                if (!Directory.Exists(targetFileParentFolder))
+                {
+                    Directory.CreateDirectory(targetFileParentFolder);
+                }
+
+                File.Copy(
+                    sourceFilePath,
+                    targetFilePath,
+                    overwrite: true);
+
+                // clear read-only bit if set
+                var fileAttributes = File.GetAttributes(targetFilePath);
+                if ((fileAttributes & FileAttributes.ReadOnly) == FileAttributes.ReadOnly)
+                {
+                    File.SetAttributes(targetFilePath, fileAttributes & ~FileAttributes.ReadOnly);
+                }
+            }
+        }
+
+        private static string EnsureTrailingSlash(string path)
+        {
+            return EnsureTrailingCharacter(path, Path.DirectorySeparatorChar);
+        }
+
+        private static string EnsureTrailingCharacter(string path, char trailingCharacter)
+        {
+            if (path == null)
+            {
+                throw new ArgumentNullException(nameof(path));
```

Comment:
Should we catch this exception somewhere? Seems like this error would make it difficult to determine that a trailing character was the issue.